### PR TITLE
DS-2880: Pubmed integration into XMLUI submission

### PR DESCRIPTION
**Pull request by rradillen**
This features allows users to search PubMed for an article during the submission process, and to automatically retrieve the metadata and fill it in into the submission form. This integration is based on the framework from https://jira.duraspace.org/browse/DS-2876.

An additional step is added to the submission process where a user can search for a PubMed article by entering the PubMed identifier, the DOI, the title or the author(s) of the publication.

From the returned results, the submitter will be able to select a single publication, from which the metadata fields will be copied into the metadata placeholders in subsequent submission form pages.

The mapping between the PubMed and DSpace metadata can be configured.

This contribution was commissioned by the Netherlands Cancer Institute.

### CODE REVIEW

File: dspace/config/xmlui.xconf [hunk 1]
```diff
@@ -182,7 +182,7 @@
         <!-- <theme name="Test Theme 2" regex="community-list" path="theme2/"/> -->
 
         <!-- Mirage theme, @mire contributed theme, default since DSpace 3.0 -->
-        <theme name="Atmire Mirage Theme" regex=".*" path="Mirage/" />
+        <theme name="Atmire Mirage Theme" regex=".*" path="Mirage2/" />
```
Comment (tdonohue) [inline dspace/config/xmlui.xconf]: Please don't change the default theme to Mirage2. This needs to be removed.

Comment (tdonohue) [inline dspace/config/xmlui.xconf]: This minor issue still remains. We should not be changing the default theme. Please remove this modification to the xmlui.xconf

**Comment by mwoodiupui** [comment]
Some notes from testing:

Is there documentation?  Does it need configuration before it will work?  I get the Publication Search page but it isn't finding anything, when given a DOI or title that I had looked up in PubMed.

We may want to have the search page commented by default?

Probably *not* the fault of this PR:  it's impossible to back out of Describe into Publication Search without filling required fields.

**Comment by mwoodiupui** [comment]
It appears that this only works with Mirage 2 due to depenency on DSpace.getTemplate() which is defined in dspace-xmlui-mirage2/src/main/webapp/scripts/utils.js only.

**Comment by jonas-atmire** [comment]
@mwoodiupui 
I've added a small bit of documentation about the current pubmed implementation, but kept this to a minimum as it works as is. 
The problem you encountered was that a maven property 'publication.url' had to be set to either an asterisk '*' or the url used in the configured service (http://eutils.ncbi.nlm.nih.gov/entrez/eutils/ by default).
There isn't a fallback implemented in the code as this would lead to possible miss-hits.

I've also added a Mirage 1 integration to use the lookup functionality (no config changes required)

**Comment by mwoodiupui** [comment]
Thank you.  The license check in dspace-xmlui/pom.xml needs an exclusion for **/handlebars/* in order to build, but this looks much better.

**Comment by mwoodiupui** [comment]
Still in Mirage 1.  I can see that the scripts are making the query and receiving a good response.  I can see the Handlebars template being fetched correctly.  But the variable to hold the compiled template is never defined.

**Comment by tdonohue** [comment]
If possible, we'd like to see any final necessary changes (noted above) completed prior to next week's DSpace Developers Meeting (Jan 28 at 15UTC). At that time, we will be (hopefully) moving rapidly towards a "feature freeze" for the 6.0 codebase.  If changes are not possible before that time, please let us know or provide us with an update prior to (or at) that meeting.

**Comment by jonas-atmire** [comment]
* Added the dataType: 'text' to the ajax return so that handlebars is able to properly compile the result.
* Changed the DSpace templates variable to use the localstore variable from javascript itself.
Mirage1 doesn't seem to play well with most custom "window."variables as they are continuously overwritten by a refresh etc.

**Comment by mwoodiupui** [comment]
Here (Firefox ESR 38.5.0) it shows an empty list of matches.  Browser console shows "TypeError:  publication_records_template is not a function".  In fact it appears to be a string which contains the definition of a function.  That makes a kind of sense to me, since localStorage is defined to store pairs of { string, string }.

**Comment by mwoodiupui** [comment]
Thank you for your patient work.  This is working now.  +1

**Comment by tdonohue** [comment]
Code looks reasonable to me (haven't had a chance to test yet). Still from a code review standpoint, I'd give this a +1

**Pull request merged**